### PR TITLE
Updating packages #2862

### PR DESCRIPTION
**Pull request by khushipatil1523**
What kind of change does this PR introduce?

This PR will Replace or update packages relying on punycode https://github.com/PalisadoesFoundation/talawa-admin/issues/2862

Issue Number:

https://github.com/PalisadoesFoundation/talawa-admin/issues/2862

Did you update for packages?

Yes

## Summary by CodeRabbit

- **Dependencies**
	- Updated multiple development dependencies to newer versions, including ESLint, Vitest, and related plugins.
	- Downgraded `jest-preview`.
	- Added new linting configuration for TypeScript files.
	- Added package override for `punycode`.

- **Chores**
	- Modified ESLint configuration to ignore Markdown files, log files, and the `talawa-admin-docs/**` directory.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -135,10 +136,10 @@
     "@types/sanitize-html": "^2.13.0",
     "@typescript-eslint/eslint-plugin": "^8.11.0",
     "@typescript-eslint/parser": "^8.5.0",
-    "@vitest/coverage-istanbul": "^2.1.5",
+    "@vitest/coverage-istanbul": "^2.1.8",
     "babel-jest": "^29.7.0",
     "cross-env": "^7.0.3",
-    "eslint": "^8.49.0",
+    "eslint": "^9.17.0",
```
[the remark this answers is not in the document]
Comment (khushipatil1523) [reply package.json]: changes done

File: package.json [hunk 2]
```diff
@@ -24,6 +24,7 @@
     "@pdfme/schemas": "^5.1.6",
     "@reduxjs/toolkit": "^2.3.0",
     "@vitejs/plugin-react": "^4.3.3",
+    "ajv": "^8.17.1",
```
[the remark this answers is not in the document]
Comment (khushipatil1523) [reply package.json]: done

**Review by palisadoes — CHANGES_REQUESTED** [review]
Why was this deleted? Are we completely using the new eslint methodology?
- `.eslintignore`

**Comment by palisadoes** [comment]
Please fix the conflicting files

**Comment by palisadoes** [comment]
Please make coderabbit.ai approve your changes

File: .eslintignore [hunk 3]
```diff
@@ -1,8 +1,10 @@
...
 # Contains the PDF file of the Tag as JSON string, thus does not need to be linted
 src/components/CheckIn/tagTemplate.ts
 package.json
 package-lock.json
 tsconfig.json
 
 # Ignore the Docusaurus website subdirectory
-docs/**
+docs/**
+
```
[the remark this answers is not in the document]
Comment (khushipatil1523) [reply .eslintignore]: done

**Comment by khushipatil1523** [comment]
> Please make coderabbit.ai approve your changes

changes done

File: .eslintignore [hunk 4]
```diff
@@ -1,8 +1,10 @@
-# Contains the PDF file of the Tag as JSON string, thus does not need to be linted
-src/components/CheckIn/tagTemplate.ts
```
Comment (palisadoes) [inline .eslintignore:2]: Why was this removed? The file is very big.

**Review by palisadoes — CHANGES_REQUESTED** [review]
See comment

File: .eslintignore (hunk 4 above)
Comment (khushipatil1523) [reply .eslintignore:2]: I deeply regret mistakenly removing the file and sincerely apologize for my oversight. I was unaware of its size and significance and will take immediate steps to rectify it.

Comment (palisadoes) [reply .eslintignore:2]: Please restore the files in the file. It's a learning process

**Review by palisadoes — CHANGES_REQUESTED** [review]
See comments

**Comment by palisadoes** [comment]
Please fix the conflicting files

**Pull request closed without merging**